lib/index.js

```diff
@@ -315,6 +316,26 @@ Socket.prototype.maxreconn  = opt('maxreconn');
 Socket.prototype.sndprio    = opt('sndprio');
 Socket.prototype.rcvprio    = opt('rcvprio');
 
+/* chan and rmchan sockopt methods. only relevant for subscription sockets */
+Socket.prototype.chan       = function(){
+  var args = arguments;
+  var channels = args.length;
+  while(--channels > -1) {
+    if(nn.Setopt(this.binding, nn.NN_SUB, nn.NN_SUB_SUBSCRIBE, args[channels]) == 0)
+      this.channels[args[channels]] = 'active';
+  };
+}
+
+Socket.prototype.rmchan   = function(channel){
+  var args = arguments;
+  var channels = args.length;
+  while(--channels > -1) {
+    if(nn.Setopt(this.binding, nn.NN_SUB, nn.NN_SUB_UNSUBSCRIBE, args[channels]) == 0)
```

Comment:
We should check that the channel is even set before calling into C++ unsubscribe.